src/encoded/audit/file.py

```diff
@@ -373,6 +373,29 @@ def audit_file_in_correct_bucket(value, system):
         )
 
 
+def audit_file_matching_md5sum(value, system):
+    '''
+    Files with md5 sums matching other files should be marked with a WARNING audit.
+    If the other files are listed as matching but in fact have different md5 sums,
+    the file should be flagged with an ERROR for incorrect metadata.
+    '''
+
+    checked_statuses = ['released', 'revoked', 'archived', 'in progress']
+    if 'matching_md5sum' not in value or value.get('status') not in checked_statuses:
+        return
+
+    for matching_file in value.get('matching_md5sum'):
```

Comment:
This is just a comment and probably a long nonsense.

It's interesting case here. So 1) this `if` above makes sure you have `matching_md5sum` and 2) the schema below makes sure you would get a list, at least an empty one, anyway. Those two then guarantee you won't get `None` from `value.get('matching_md5sum')` which will trigger error for the `for` loop. Personally, I don't want to have a code depends on code at other places or far away. Like if schema changed or something, there is no guarantee you won't get `None` or trigger error. So I like the `get` here but you could also have the default being list. On the contrary, though, an error here could be good to tell whoever changes codes in the future to check back here. So I don't really know and I'm not really recommending further changes here.

Reply:
The audit test would fail if the schema got updated, so I think this is OK. The developer would know that they need to update the audit.